index.rst

```diff
@@ -1,29 +1,299 @@
+Butler is a special case since the Butler library itself will directly load the top-level configuration and can be configured (via an environment variable) with its location, and thus doesn't need a separate helper library.
+
+Implementation
+--------------
+
+The current implementation of this helper library lives in two packages: lsst.rsp_ and lsst_efd_client_.
+The former currently provides ways to access the TAP service (``lsst.rsp.get_tap_service`` and ``lsst.rsp.retrieve_query``).
+The latter wraps the current EFD discovery service and provides methods to perform various types of EFD queries.
+
+.. _lsst.rsp: https://github.com/lsst-sqre/lsst-rsp
+.. _lsst_efd_client: https://efd-client.lsst.io/
+
+We expect to want to add support for additional services to the lsst.rsp package, similar to the support for the TAP service.
+The exact details may depend on the service; there are a few options for what may be most convenient for the user:
+
+- A mechanism to obtain the URL of the service, leaving all other details including authentication to the user.
+  This gives the user the most flexibility, at the cost of only wrapping the discovery step and leaving authentication and all other details of service access to the user.
+  This is the approach taken by the existing ``lsst.rsp.get_tap_service``.
+
+- A way to initiate a third-party client with the details required to access a service.
+  For example, the helper function may return a PyVO_ client object with the service URL and authentication credentials already configured, or initiate the standard operation in a third-party client and return the results.
+  This is the approach taken by the existing ``lsst.rsp.retrieve_query``.
 
-The Rubin Science Platform is composed of a large number of separate services, some following external standards such as IVOA and some providing internal protocols. The services available and the data behind them will vary by site and need to be discoverable by users. This tech note lists known discovery needs for both services and schemas and proposes an implementation plan for meeting those needs.
+- Provide a full-fledged client for the service, which manages the connection and authentication and provides methods to perform all supported operations for that service.
+  This is the approach taken by lsst_efd_client_.
 
-Add content here
-================
+There is no one solution that makes sense for every service.
+Which approach we take for a service depends on the details of that service, the availability of third-party libraries, and the anticipated use cases of users.
 
-Add content here.
-See the `reStructuredText Style Guide <https://developer.lsst.io/restructuredtext/style.html>`__ to learn how to create sections, links, images, tables, equations, and more.
+However, we can support, for every service, the first, most basic level of support: return the URL of the named service in the local Science Platform, if it exists.
```

Comment:
Not that it matters, but this does open the door for "give me all services you know of on the local Science Platform" which could be useful. In any case it would be good to standardize to lsst.rsp.get_service("tap"), lsst.rsp.get_service("efd") etc rather than have unique calling patterns for reach service

Reply:
Good point, I'll say that explicitly and also mention asking for the list.